### DISCUSSION TITLE
XAML Localization update

### DISCUSSION THREAD

**Discussion by gevart** [unoplatform/uno.extensions#1781] [Q&A]
Hello everyone.

I'm using `Uno.Extentions.Localization` and need to allow user to select UI culture during the first run. So I used `ILocalizationService` mentioned in the docs and it's working fine.
But the docs also says it requires an app restart to apply the changes and that's true, after calling `ILocalizationService.SetCurrentCultureAsync` only in-code references of `IStringLocalizer` get new culture strings, XAML localization via `x:Name` keeps the old culture until I restart the app.
I've also tried to navigate to a couple of other pages (with `NavigationCacheMode="Disabled"`) after changing culture and result is the same: in-code localization gets new culture strings while XAML localization keeps the old culture strings.

So the questions are:
- can I do something to make XAML localization to reload the strings for the new culture without restart?
- or if not, can I programmatically restart my app? If it can't be done universally, number 1 priority for me are Mobile and Windows targets.

Thanks for your time.

**Comment by MartinZikmund** [comment]
I think @nickrandolph might know on this one. I would expect that unloading and reloading UI would make the new localizations to take effect

**Reply by gevart** [reply]
Not sure about unloading, but I've tried to navigate to different pages with disabled NavigationCacheMode and it didn't work for XAML strings, they were still in the old culture.

**Reply by MartinZikmund** [reply]
That should work indeed, @Xiaoy312 do you have ideas?

**Comment by gevart** [comment]
It appears that https://github.com/unoplatform/Uno.Samples/tree/master/UI/LocalizationSamples/RuntimeCultureSwitching is not currently working for Windows head either.
